### ISSUE TITLE
Inconsistent database naming capitalization

### ISSUE THREAD

**Issue by davidjanney** [jhipster/generator-jhipster#4668]
##### **Overview of the issue**

The database names between the generated docker files and the application-dev.yml and application-prod.yml have inconsistent capitalization. This should have the same capitalization for case-sensitive file systems.

##### **Motivation for or Use Case**

When running the generator and starting up the db (mysql in this example) via docker, the Java application won't be able to connect to the database on case-sensitive systems since the capitalization is inconsistent.

##### **Reproduce the error**

Run the generator with 
[yo-rc.txt](https://github.com/jhipster/generator-jhipster/files/646901/yo-rc.txt)

Build and run the application with docker:
./mvnw package -Pprod docker:build
docker-compose -f src/main/docker/app.yml up

##### **Related issues**

Didn't find any.

##### **Suggest a Fix**

Should change <%= baseName %> to <%= baseName.toLowerCase() %> in a few lines at
https://github.com/jhipster/generator-jhipster/blob/b3ba0b83b2639a69793840d58932a79a91813df8/generators/server/templates/src/main/resources/config/_application-prod.yml

as well as

https://github.com/jhipster/generator-jhipster/blob/b3ba0b83b2639a69793840d58932a79a91813df8/generators/server/templates/src/main/resources/config/_application-dev.yml

##### **JHipster Version(s)**

Using version 3.12.1

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

See the yo-rc.txt file above.

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

No generated entities.

##### **Browsers and Operating System**

OSX

**Comment by ruddell** [comment]
I didn't see any issue on Mac or Ubuntu 16 (which has a case-sensitive file system).  

Our docker-compose mysql.yml uses the flag `lower_case_table_names` when starting up the MySQL docker image which I think prevents this issue.  From [the docs](http://dev.mysql.com/doc/refman/5.7/en/identifier-case-sensitivity.html): "Table names are stored in lowercase on disk and name comparisons are not case sensitive. MySQL converts all table names to lowercase on storage and lookup. This behavior also applies to database names and table aliases."

**Comment by pascalgrimaud** [comment]
As @ruddell, I don't see any issue

Playing with your [.yo-rc.json](https://github.com/pascalgrimaud/jhipster-travis-tester/blob/4668-capitalization-database-name/app/.yo-rc.json)
Launching the same [commands](https://github.com/pascalgrimaud/jhipster-travis-tester/blob/4668-capitalization-database-name/app/test.sh#L28-L29)
See the full log in Travis: https://travis-ci.org/pascalgrimaud/jhipster-travis-tester/builds/183383738

**Comment by jdubois** [comment]
Yes, for me this is a problem we already had, and it was supposed to be solved.... But here @davidjanney talks just about the database names, not the table names, and maybe we just solved it for table names? Then again, how come we don't have the same issue on Mac/Linux/Docker?

I don't understand why we can't reproduce the issue here, @davidjanney is using Docker so @ruddell and @pascalgrimaud should have the same problem...

**Comment by ruddell** [comment]
@jdubois `lower_case_table_names` affects both table and database names

**Comment by jdubois** [comment]
OK thanks @ruddell  , so this is supposed to be solved, and we can't reproduce, that's why I'm closing this.
@davidjanney if you have more information and find a way to reproduce, please add comments, I'm really wondering what happened here.

**Comment by davidjanney** [comment]
@jdubois @ruddell @pascalgrimaud I apologize, it looks like I pulled the yo-rc file from the wrong test.  Websocket should have been set to "spring-websocket" not false.

I'm trying to run a few more tests to make sure that was really the difference.

Earlier when I was testing this, I kept everything as is and ran using docker, and got the following error output:

[couldnotfinddbname.txt](https://github.com/jhipster/generator-jhipster/files/647308/couldnotfinddbname.txt)

After modifying the database name in the config file to be lowercase, I get no errors and everything ran fine.

I'll post the yo-rc file once I verify I'm pulling the correct one. It's just taking a few iterations here. Spent a little while on this today.
